test/src/test/java/jenkins/security/BasicHeaderProcessorTest.java

```diff
@@ -155,4 +188,79 @@ public HttpResponse doIndex() {
             return HttpResponses.plainText(u!=null ? u.getId() : "anonymous");
         }
     }
+    
+    @TestExtension
+    public static class DummySecurityListener extends SecurityListener {
+        final EventQueue<UserDetails> authenticatedCalls;
+        final EventQueue<String> failedToAuthenticateCalls;
+        final EventQueue<String> loggedInCalls;
+
+        public DummySecurityListener(){
+            this.authenticatedCalls = new EventQueue<>();
```

Comment:
can be just initialized in the definition, and then you could keep a default constructor

Reply:
done